lib/CppInterOp/CppInterOp.cpp

```diff
@@ -855,8 +855,21 @@ static void GetClassDecls(TCppScope_t klass,
     if (auto* MD = dyn_cast<DeclType>(DI))
       methods.push_back(MD);
     else if (auto* USD = dyn_cast<UsingShadowDecl>(DI))
-      if (auto* MD = dyn_cast<DeclType>(USD->getTargetDecl()))
-        methods.push_back(MD);
+      if (auto* MD = dyn_cast<DeclType>(USD->getTargetDecl())) {
+        if (auto* CUSD = dyn_cast<ConstructorUsingShadowDecl>(DI)) {
+          if (auto* CXXCD =
+                  dyn_cast_or_null<CXXConstructorDecl>(CUSD->getTargetDecl())) {
+            CXXConstructorDecl* Result = getSema().findInheritingConstructor(
+                SourceLocation(), CXXCD, CUSD);
+            // Result is appended to the decls, i.e. CXXRD, iterator
```

Comment:
The implementation is awkward... Would that mean that the iterators are invalidated?